From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quentin Perret <qperret@google.com>
Date: Tue, 24 Sep 2019 17:28:02 +0100
Subject: ANDROID: Fix arm64 allmodconfig build

Allmodconfig on arm64 enables CPU_BIG_ENDIAN=y, which causes issues with
ld.lld which doesn't support linking aarch64be-linux-gnu targets (see
https://reviews.llvm.org/D58655#1410281). However, it is very unlikely
that real android devices run with arm64 BE hardware in practice. So,
until we can find a better fix, let's simply force CPU_BIG_ENDIAN=n for
allmodconfig builds.

Bug: 141733632
Bug: 140224784
Signed-off-by: Quentin Perret <qperret@google.com>
Change-Id: Ic4693ae1f462144c8219b397463ca341f6fe08a1
---
 build.config.allmodconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build.config.allmodconfig b/build.config.allmodconfig
index 6e0404b8de47..deaa8f3d5a69 100644
--- a/build.config.allmodconfig
+++ b/build.config.allmodconfig
@@ -6,7 +6,8 @@ POST_DEFCONFIG_CMDS="update_config"
 function update_config() {
     ${KERNEL_DIR}/scripts/config --file ${OUT_DIR}/.config \
          -d TEST_KMOD  \
-         -d XFS_FS
+         -d XFS_FS \
+         -d CPU_BIG_ENDIAN
     (cd ${OUT_DIR} && \
      make O=${OUT_DIR} $archsubarch CC=${CC} CROSS_COMPILE=${CROSS_COMPILE} olddefconfig)
 }
